### PR TITLE
Introduce kernel-uvm package

### PR DESCRIPTION
**Pull request by christopherco**
###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [x] The toolchain has been rebuilt successfully (or no changes were made to it)
- [x] The toolchain/worker package manifests are up-to-date
- [x] Any updated packages successfully build (or no packages were changed)
- [x] Packages depending on static components modified in this PR (Golang, `*-static` subpackages, etc.) have had their `Release` tag incremented.
- [x] Package tests (%check section) have been verified with RUN_CHECK=y for existing SPEC files, or added to new SPEC files
- [x] All package sources are available
- [x] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/tools/cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`, `.github/workflows/cgmanifest.json`)
- [x] LICENSE-MAP files are up-to-date (`./SPECS/LICENSES-AND-NOTICES/data/licenses.json`, `./SPECS/LICENSES-AND-NOTICES/LICENSES-MAP.md`, `./SPECS/LICENSES-AND-NOTICES/LICENSE-EXCEPTIONS.PHOTON`)
- [x] All source files have up-to-date hashes in the `*.signatures.json` files
- [x] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [x] Documentation has been updated to match any changes to the build system
- [x] Ready to merge

---

###### Summary 

In order to support kata-container's lightweight VMs, a guest VM kernel is needed. For our purposes, we want a very minimal kernel that uses our Mariner Linux kernel source tree and is specifically configured to work well within this lightweight VM environment.

This change introduces a new `kernel-uvm` RPM package which builds and packages a kernel binary specifically for lightweight "utility" VMs.

This kernel's config file is initially sourced from the kata-containers releases, specifically `kata-static-2.5.0-x86_64.tar.xz` under `opt/kata/share/kata-containers/config-5.15.48`

###### Change Log  

- Add kernel-uvm spec
- Add cgmanifest entry for kernel-uvm
- Add licenses entry for kernel-uvm
- kernel-uvm: add Xen note to KCFLAGS 
  ```
  When attempting to boot this guest kernel with kata + cloud-hypervisor,
  we encountered the following error:
  
     Warning  FailedCreatePodSandBox  12s (x2 over 31s)  kubelet
     Failed to create pod sandbox:
     rpc error: code = Unknown desc = failed to create containerd task:
     failed to create shim task: error: 500
     reason: ApiError(VmBoot(KernelMissingPvhHeader)): unknown
  
  This error occurs because the kernel binary does not have the Xen note
  in the ELF notes.
  
     $ readelf -n boot/vmlinux.bin
  
     Displaying notes found in: .notes
       Owner                 Data size       Description
       GNU                  0x00000020       NT_GNU_PROPERTY_TYPE_0
           Properties: <procesor-specific type 0xc0010001 data: cf 00 00 00 >
             <procesor-specific type 0xc0010002 data: 07 00 00 00 >
       GNU                  0x00000014       NT_GNU_BUILD_ID (unique build ID bitstring)
         Build ID: 1063b1721e10386f12c548ca2d0e79d157cb0665
     readelf: Warning: note with invalid namesz and/or descsz found at offset 0x58
     readelf: Warning:  type: 0x756e694c, namesize: 0x00000004, descsize: 0x00000101, alignment: 8
  
  Our assembler has the mx86-used-note=yes
  
     # as -help | grep used-note
       -mx86-used-note=[no|yes] (default: yes)
  
  so supply -mx86-used-note=no to the kernel build to add the Xen note.
     $ readelf -n vmlinux.bin
  
     Displaying notes found in: .notes
       Owner                 Data size       Description
       GNU                  0x00000014       NT_GNU_BUILD_ID (unique build ID bitstring)
         Build ID: 6988772c557469f4f0f8fad76525c13f60782cec
       Linux                0x00000004       func
        description data: 00 00 00 00
       Linux                0x00000001       OPEN
        description data: 00
       Xen                  0x00000008       Unknown note type: (0x00000012)
  
  Signed-off-by: Chris Co <chrco@microsoft.com>
  ```

###### Does this affect the toolchain?  

**NO**

###### Test Methodology

Local build

### CODE REVIEW

File: SPECS/kernel-uvm/kernel-uvm.spec [hunk 1]
```diff
@@ -0,0 +1,1109 @@
...
+
+%ifarch x86_64
+%define arch x86_64
+%define archdir x86
+%define config_source %{SOURCE1}
+%endif
+
+Summary:        Linux Kernel for Kata UVM
+Name:           kernel-uvm
+Version:        5.15.48.1
```
Comment (oliviacrain) [inline SPECS/kernel-uvm/kernel-uvm.spec:13]: Is the version of `kernel-uvm` supposed to match that of the `kernel`/`kernel-headers` packages? If so, please modify the contents of the `version_matching_groups` variable in `toolkit/scripts/check_entangled_specs.py` to reflect that, so we can use our PR checks to keep these packages in sync.

Comment (rlmenge) [reply SPECS/kernel-uvm/kernel-uvm.spec:13]: Spoke with Chris offline. The kernel configs for kernel-uvm come from an outside source which is not in sync with the latest LTS version. This is similar to how kernel-rt is dependent on a patchset sync'd with a specific version of the kernel which may not be updated to LTS's version. Therefore, kernel-uvm and kernel-rt may be behind as we wait for those sources to become available and/or update the kernel to address CVEs.

Comment (oliviacrain) [reply SPECS/kernel-uvm/kernel-uvm.spec:13]: Sounds great, thanks Rachel!

File: SPECS/kernel-uvm/kernel-uvm.spec [hunk 2]
```diff
@@ -0,0 +1,1109 @@
...
+Release:        6%{?dist}
```
Comment (neha170) [inline SPECS/kernel-uvm/kernel-uvm.spec:14]: Nit: Release

**Review by neha170 — APPROVED** [review]

File: SPECS/kernel-uvm/kernel-uvm.spec (hunk 2 above)
Comment (christopherco) [reply SPECS/kernel-uvm/kernel-uvm.spec]: Thanks for catching this! Fixed in newest push

File: SPECS/kernel-uvm/kernel-uvm.spec (hunk 1 above)
Comment (christopherco) [reply SPECS/kernel-uvm/kernel-uvm.spec:13]: What Rachel wrote is true, especially for right now.
Depending on how things progress, we may need to deviate `kernel-uvm` from the kernel configs posted by the upstream kata releases. Meaning we may need to revisit syncing with the LTS version.

**Review by danmihai1 — APPROVED** [review]

**Review by neha170 — COMMENTED** [review]
Hi, we can discuss needing to make some changes to uvm kernel rpm, so that kata script is able to generate an initrd for it, thanks

**Pull request merged**